components/ai_chat/core/browser/ai_chat_database.cc

```diff
@@ -0,0 +1,914 @@
+        "DELETE FROM conversation_entry_event_completion WHERE "
+        "conversation_entry_uuid=?"));
+    delete_statement.BindString(0, conversation_entry_uuid);
+    if (!delete_statement.Run()) {
+      DLOG(ERROR)
+          << "Failed to delete from conversation_entry_event_completion "
+             "for id: "
+          << conversation_entry_uuid;
+      return false;
+    }
+  }
+
+  // Delete from conversation_entry_event_search_queries
+  {
+    static const std::string kQuery =
+        "DELETE FROM conversation_entry_event_search_queries WHERE "
+        "conversation_entry_uuid=?";
+    sql::Statement delete_statement(GetDB().GetUniqueStatement(kQuery));
+    CHECK(delete_statement.is_valid());
+    delete_statement.BindString(0, conversation_entry_uuid);
+    if (!delete_statement.Run()) {
+      DLOG(ERROR) << "Failed to delete from "
+                     "conversation_entry_event_search_queries for conversation "
+                     "entry uuid: "
+                  << conversation_entry_uuid;
+      return false;
+    }
+  }
+
+  // Delete edits
+  {
+    static const std::string kQuery =
+        "DELETE FROM conversation_entry WHERE editing_entry_uuid = ?";
+    sql::Statement delete_statement(GetDB().GetUniqueStatement(kQuery));
+    CHECK(delete_statement.is_valid());
+    delete_statement.BindString(0, conversation_entry_uuid);
+    if (!delete_statement.Run()) {
+      DLOG(ERROR) << "Failed to delete from conversation_entry for "
+                     "conversation entry uuid: "
+                  << conversation_entry_uuid;
+      return false;
+    }
+  }
+
+  // Delete from conversation_entry
+  {
+    static const std::string kQuery =
+        "DELETE FROM conversation_entry WHERE uuid=?";
+    sql::Statement delete_statement(GetDB().GetUniqueStatement(kQuery));
+    CHECK(delete_statement.is_valid());
+    delete_statement.BindString(0, conversation_entry_uuid);
+    if (!delete_statement.Run()) {
+      LOG(ERROR) << "Failed to delete from conversation_entry for id: "
+                 << conversation_entry_uuid;
+      return false;
+    }
+  }
+
+  if (!transaction.Commit()) {
+    DVLOG(0) << "Transaction commit failed with reason: "
+             << db_.GetErrorMessage();
+    return false;
+  }
+  return true;
+}
+
+bool AIChatDatabase::DeleteAllData() {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  static const std::string kDropConversationTableQuery =
+      "DROP TABLE conversation";
+  static const std::string kDropAssociatedContentTableQuery =
+      "DROP TABLE associated_content";
+  static const std::string kDropConversationEntryTableQuery =
+      "DROP TABLE conversation_entry";
+  static const std::string kDropCompletionEventTableQuery =
+      "DROP TABLE conversation_entry_event_completion";
+  static const std::string kDropSearchQueriesEventTableQuery =
+      "DROP TABLE conversation_entry_event_search_queries";
+
+  return GetDB().Execute(kDropConversationTableQuery) &&
```

Comment:
I do have all the unit tests run both with the creation-time schema and the post-DeleteAllData schema. `sql::Database::Delete(FilePath)` does exist though, so why not. I think I was trying to avoid having to handle issues closing the handles and deleting the file which sometimes come up on a random platfform 🤷

Reply:
I've now changed this to use `Raze()` which wipes everything